### ISSUE TITLE
[feature] support enable/disable scaling dynamically

### ISSUE THREAD

**Issue by fredwangwang** [hashicorp/nomad#10117]
### Proposal
Add an api endpoint to dynamically enable or disable a specific group level scaling.

### Use-cases
In case an emergent event happens and the operator decide the best action is to scale up the instance count. However due to the autoscaler is still enabled, the operator needs to constantly wrestling against the autoscaler.

During such event the operator should be able to disable autoscaling for a specific group and use manual scaling instead without needing to redeploy the job.

### Potential issues
Authority of the enable/disable status. It might need something similar to `-preserve-counts` to determine whether to reset the scaling status during a redeploy.

**Comment by drewbailey** [comment]
Hey @fredwangwang, thanks for taking the time to submit this feature request.

I think you should be able to set `enable=false` on the scaling policy, have you tried to see if that fulfills your use case? If it doesn't, any details where it falls short would be great, thanks!

**Comment by fredwangwang** [comment]
Hi @drewbailey Thanks for the reply. 

> without needing to redeploy the job.

I think that is the **key** I wanted this feature. I understand setting the variable in the scaling stanza could do it, but it does need a job redeploy isnt it

**Comment by drewbailey** [comment]
@fredwangwang you are correct, there would be a new deployment, although it's an in-place update so the allocations wouldn't be rescheduled.

**Comment by mikenomitch** [comment]
After discussing with the team, I'm going to close this out.

We aren't sure that this is a good idea. Ideally you would just change the job to disable autoscaling. I can see a case where this would be helpful but I don't think it will come up often enough to justify adding this feature to be honest.

Since I can't see us prioritizing this for both of these reasons, going to close this out to clean up open tickets.

In any case, we appreciate the feature request! If others feel like this would be very useful, please do comment/let us know.

**Comment by pop** [comment]
I have a similar use-case where we want to temporarily scale *down* a Group which is part of a larger Job and is which is managed by the Nomad Autoscaler. It would be nice to forcibly scale down the Group without the autoscaler scaling it back up automatically in this case. All deployments go through a CI/CD pipeline making the cost of trivial changes like this fairly high.

Being able to edit a scaling policy via the API and/or CLI would help achieve our "scale down to 0 for a while" use case by first disabling the autoscaling policy, then scaling the group down, and finally re-enabling the autoscaling policy.

That said, applying the change in code is nice as it gives us a paper trail of when the policy was modified, so there's no surprises about the current state of autoscaling for this Job (i.e., "The Job Deployment says the Policy is enabled but the Policy itself says it is disabled?!")

TLDR it would be nice to disable autoscaling policies dynamically, but the consistency of forcing all changes through a deployment is nice.
